### PR TITLE
[14.0][MIG] pos_customer_required: Migration to 14.0

### PR DESCRIPTION
**Pull request by KNVx**

### CODE REVIEW

**Comment by KNVx** [comment]
Migration to 14.0

Please @eantones can you review it?

**Comment by eantones** [comment]
LGTM!

File: pos_customer_required/static/src/js/ProductScreen.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+   Copyright NuoBiT Solutions - Kilian Niubo <kniubo@nuobit.com>
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl) */
+odoo.define("pos_customer_required.ProductScreen", function (require) {
+    "use strict";
+
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const Registries = require("point_of_sale.Registries");
+
+    var PCRProductScreen = (ProductScreen) =>
+        class PCRProductScreen extends ProductScreen {
```
Comment (legalsylvain) [inline pos_customer_required/static/src/js/ProductScreen.js]: ``        class extends ProductScreen {``
should be enought. (also in Payment screen).

(see other v14.0 module like (``pos_margin``, ``pos_access_right``)

**Review by legalsylvain — COMMENTED** [review]
one minor review. LGTM otherwise.
thanks for porting & adapt this module.

File: pos_customer_required/static/src/js/ProductScreen.js (hunk 1 above)
Comment (KNVx) [reply pos_customer_required/static/src/js/ProductScreen.js]: Done!
Thanks @legalsylvain

**Review by eantones — APPROVED** [review]
LGTM!

**Review by dsolanki-initos — APPROVED** [review]
Functionally test and code review. looks good to me.

**Comment by dsolanki-initos** [comment]
Can we merge this PR?

**Comment by dsolanki-initos** [comment]
@legalsylvain can we merge this PR?

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @dsolanki-initos : I tested on runboat, and it doesn't work. there is an error, when selecting a product. 

looking the log, it seems that there is a problem on the module pos_reset_search you migrated here : https://github.com/OCA/pos/pull/651

Could you help to solve the problem ? (https://github.com/OCA/pos/issues/705).

here is the pending PR : https://github.com/OCA/pos/pull/749

thanks.

**Comment by dsolanki-initos** [comment]
@ivantodorovich can we merge this PR?

**Comment by legalsylvain** [comment]
> @ivantodorovich can we merge this PR?

@dsolanki-initos : could you answer to questions, before asking work from OCA members ? 

runboat is still failing.

**Comment by dsolanki-initos** [comment]
@ivantodorovich can we merge this PR?

**Comment by dsolanki-initos** [comment]
Helllo @ivantodorovich Can we merge this PR?

**Comment by ivantodorovich** [comment]
/ocabot rebase


@legalsylvain , is it ok for your comments?

**Comment by legalsylvain** [comment]
Dont have time to test it again. If it's ok for you, feel free to merge.

**Comment by ivantodorovich** [comment]
@KNVx can you please rebase?

runboat is still broken due to this fix not being integrated here:
- https://github.com/OCA/pos/pull/749

**Comment by dsolanki-initos** [comment]
@KNVx can you please rebase ?

**Comment by dsolanki-initos** [comment]
@KNVx can you please follow the necessary action?

**Comment by ivantodorovich** [comment]
@dsolanki-initos , please try not to be so insistent. Most of us are busy and that's ok.
If you're in a real hurry, you can always make a new PR continuing his work 😉

**Comment by dsolanki-initos** [comment]
@KNVx Can you please rebase?

**Comment by eantones** [comment]
Done @dsolanki-initos :-)

**Comment by dsolanki-initos** [comment]
Can we merge this PR?

**Comment by dsolanki-initos** [comment]
@ivantodorovich can we merge this PR?

**Review by ivantodorovich — APPROVED** [review]
Tested on runboat and it's working now 👍🏻 
Thanks

**Pull request merged**